src/jacobians.jl

```diff
@@ -119,10 +123,11 @@ function finite_difference_jacobian!(J::AbstractMatrix,
     f_in       :: Union{T2,Nothing}=nothing;
     relstep=default_relstep(fdtype, eltype(x)),
     absstep=relstep,
-    color = eachindex(x)) where {T1,T2,T3}
+    color = eachindex(x),
+    sparsity = nothing) where {T1,T2,T3}
```

Comment:
We should have a better default here, and assume that we have sparsity if `J` is a sparse matrix.

Reply:
`J isa SparseMatrixCSC?J:nothing` ?

Reply:
Yeah that sounds good for now. In the near future we'll want to handle structured matrices https://github.com/JuliaDiffEq/DiffEqDiffTools.jl/issues/67 , but for now it would error anyways.